src/app/frontend/deploy/deploy_stateconfig.js

```diff
@@ -44,3 +46,23 @@ function resolveNamespaces($resource) {
 
   return resource.get().$promise;
 }
+
+/**
+ * @param {!angular.$resource} $resource
+ * @return {!angular.Resource<!backendApi.ReplicaSetSpec>}
+ * @ngInject
+ */
+function getProtocolsResource($resource) {
+  return $resource('/api/appdeployments/protocols');
+}
+
+/**
+ * @param {!angular.Resource<!backendApi.Protocols>} protocolsResource
+ * @returns {!angular.$q.Promise}
+ * @ngInject
+ */
+function getDefaultProtocols(protocolsResource) {
+  /** @type {!backendApi.ProtocolsSpec} */
+  let protocolsSpec = {isExternal: false};
```

Comment:
I see 2 solutions:

1. Add locals `isExternal` to state config, but then propagating this would require adding yet another binding to deploy from settings directive.
2. Create export const in deploy from settings controller then import and use in here.

I actually like first option better. What do you think?